common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -1097,8 +1101,21 @@ public UTF8String copy() {
   @Override
   public int compareTo(@Nonnull final UTF8String other) {
     int len = Math.min(numBytes, other.numBytes);
-    // TODO: compare 8 bytes as unsigned long
-    for (int i = 0; i < len; i ++) {
+    int words = len / Longs.BYTES;
```

Comment:
While we're optimizing -- you only ever use the value of this as multiplied by 8. You could just define something like `wordMax` as `(len / 8) * 8` and avoid the multiplication below.